### PR TITLE
ContextualMenuSplitButton: Adding the correct attributes for screen readers to read the button correctly

### PR DESCRIPTION
**Pull request by chang47**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

**Problem:** 
Currently split buttons in the contextual menu's aren't being read properly with Screen Readers like Narrator.

Instead of reading: <name/aria-label of button>, :<collapsed/expanded> menuitem

the screen reader reads:

button

**Solution:**

I:
* added aria-label instead of aria-labelledby, which will allow the screen reader to read the text correctly
* added aria-expanded so the screen reader will understand that the menu is opened or not
* changed the role of the container to be a menuitem and not a button

#### Focus areas to test

I validated in the Contextual Menu page that Narrator now reads what we expect for split button items.
I also validated in my own individual app that Narrator now reads the expected text.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -640,6 +640,9 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       ev.preventDefault();
       ev.stopPropagation();
     }
+
+    // Note: When enter is pressed, we will let the event continue to propagate
```
Comment (chang47) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Weird, I'm not sure why these changes are showing up. I'm going to remove them.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx [hunk 2]
```diff
@@ -56,10 +59,11 @@ export class ContextualMenuSplitButton extends ContextualMenuItemWrapper {
           <div
             data-ktp-target={ keytipAttributes['data-ktp-target'] }
             ref={ (splitButton: HTMLDivElement) => this._splitButton = splitButton }
-            role={ 'button' }
-            aria-labelledby={ item.ariaLabel }
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx]: why are you getting rid of aria-labelledby?

Comment (chang47) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx]: As you mentioned, there is no way to pass it in currently.  We're not currently even using it correctly. We're just passing it the label which should be a text and not a reference to another element that should display the label.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx [hunk 3]
```diff
@@ -56,10 +59,11 @@ export class ContextualMenuSplitButton extends ContextualMenuItemWrapper {
...
+            role={ 'menuitem' }
+            aria-label={ item.ariaLabel ? item.ariaLabel : item.name }
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx]: do we want to default to name? Or would we rather just use ariaLabel if it's there, who knows what someone may but on the name since it's not supposed to be used for anything accessibility related

Comment (chang47) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuSplitButton.tsx]: I got rid of it. I swore I saw somewhere in the definition for aria-label that says that we should use name be default if aria-label is empty. Looking back at it I misunderstood it.
